rfcs/0000-policy_target_scope.md

```diff
@@ -0,0 +1,267 @@
+# RFC A single Policy scoped to a HTTPRouteRule
+
+- Feature Name: Policy targeting HTTPRoute and HTTPRouteRules
+- Start Date: 2022-11-09
+- RFC PR: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/pull/0000)
+- Issue tracking: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Currently both `RateLimitPolicy` and `AuthPolicy` allow for a policy to only affect a _subset_ of the
+traffic on a targeted `HTTPRoute`. This proposal aims at simplifying the scope of a policy to target
+_all_ the traffic of a route or the traffic specified by a specific `HTTPRouteRule`'s `HTTPRouteMatch`.
+
+# Motivation
+[motivation]: #motivation
+
+As per the specification, a given resource will always need to support multiple policies targeting it.
+Even if we only support a single e.g. `RateLimitPolicy` per resource, there will alway be a possibility
+for another policy being defined else where in that resource's hierarchy: e.g. _policy1_ targeting the
+`Gateway`, while _policy2_ targeting the `HTTPRoute` being attached to the `Gateway`. Until there is an
+`HTTPRoute` tho, _policy1_ will have no semantic, as there is no traffic routed, so no policy for
+Kuadrant to enforce, as it operates on the data plane. When an `HTTPRoute` is attached to the `Gateway`
+is the point when that _policy1_ becomes applicable. If that `HTTPRoute` also defines a `RateLimitPolicy`,
+only one will be enforced, according to the `default` & `override` rules of the specification. 
+
+The idea is to keep things simple (if only for now), so that when traffic is routed through an
+`HTTPRoute` only one policy of a `kind` becomes applicable and applies to the whole traffic of that route
+(or a subset thereof, see below).
+
+The user might want to qualify traffic to only a specific `HTTPRouteRule` of a given `HTTPRoute`. That would be
+permitted for as long as the `Policy` reflects the exact same set of `HTTPRouteMatch` as the `HTTPRouteRule` defines.
+If two policies target the same `HTTPRouteRule`, one would take precedence over the other according to
+the Gateway API's merging rules.
+
+With these rules, Kuadrant can "easily" identify what policies apply to an `HTTPRoute`, possibly
```

Comment:
one thing that occurred to me might be to actually reflect this in the status of the HTTPRoute in a condition